spackbot/handlers/style.py

```diff
@@ -31,90 +32,22 @@ async def style_comment(event, gh):
-
-    # Tell the user the style fix is going to take a minute or two
-    message = "Let me see if I can fix that for you! This might take a moment..."
-    await gh.post(event.data["issue"]["comments_url"], {}, data={"body": message})
-
-    # Get the username of the original committer
-    user = pr["user"]["login"]
-
-    # We need the user id if the user is before July 18. 2017
-    email = await helpers.get_user_email(gh, user)
-
-    # We need to use the git url with ssh
-    branch = pr["head"]["ref"]
-    full_name = pr["head"]["repo"]["full_name"]
-    fork_url = f"git@github.com:{full_name}.git"
-
-    # At this point, we can clone the repository and make the change
-    with helpers.temp_dir() as cwd:
-
-        # Clone a fresh spack develop to use for spack style
-        git("clone", helpers.spack_upstream, "spack-develop")
-
-        spack = sh.Command(f"{cwd}/spack-develop/bin/spack")
-
-        # clone the develop repository to another folder for our PR
-        git("clone", "spack-develop", "spack")
-
-        os.chdir("spack")
-        git("config", "user.name", user)
-        git("config", "user.email", email)
-
-        # This will authenticate the push with the added ssh credentials
-        git("remote", "add", "upstream", helpers.spack_upstream)
-        git("remote", "set-url", "origin", fork_url)
-
-        # we're on upstream/develop. Fetch and check out just the PR branch
-        git("fetch", "origin", f"{branch}:{branch}")
-        git("checkout", branch)
-
-        # Save the message for the user
-        res, err = helpers.run_command(spack, ["--color", "never", "style", "--fix"])
-        message = comments.get_style_message(res)
-
-        # Commit (allow for no changes)
-        res, err = helpers.run_command(
-            git,
-            ["commit", "-a", "-m", f"[spackbot] updating style on behalf of {user}"],
-        )
-
-        # Continue differently if the branch is up to date or not
-        if is_up_to_date(res):
-            message += "\nI wasn't able to make any further changes, but please see the message above for remaining issues you can fix locally!"
-            return message
-        message += "\nI've updated the branch with isort fixes."
-
-        # Finally, try to push, update the message if permission not allowed
-        try:
-            git("push", "origin", branch)
-        except Exception:
-            message += "\n\nBut it looks like I'm not able to push to your branch. 😭️ Did you check maintainer can edit when you opened the PR?"
-
-    return message
+    auth_method = {}
+
+    if "installation_id" in kwargs:
+        auth_method["installation_id"] = kwargs["installation_id"]
+    elif "oauth_token" in kwargs:
+        auth_method["oauth_token"] = kwargs["oauth_token"]
+
+    task_q = work_queue.get_queue()
+    fix_style_job = task_q.enqueue(
+        fix_style_task,
+        event,
+        auth_method,
+        job_timeout=WORKER_JOB_TIMEOUT,
```

Comment:
Is a comment made on the PR if the job times out?  I think it should be.

Reply:
Agreed.  The `rq` docs say if job exceeds timeout, it is terminated and marked as failed, so I added an `on_failure` callback to handle this, which also just makes sense in general.  Tested in the dev environment, you can see how it looks live [here](https://github.com/spack-test/spack/pull/17#issuecomment-1182562079).